### PR TITLE
[FLOSS H2] Added documentation on how to run all tests with podman instead of docker

### PR DESCRIPTION
**Pull request by muskater**
Podman is an open source, daemonless container engine and can be used as drop in replacement for docker. However, some things like volume mounts work a little bit different. I added to the [run all tests with docker tutorial](https://www.libelektra.org/tutorials/run-all-tests-with-docker) alternative instructions on how to run the container with podman.

## Basics

These points need to be fulfilled for every PR:

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [ ] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

File: doc/tutorials/run_all_tests_with_docker.md [hunk 1]
```diff
@@ -65,12 +81,24 @@ Afterwards pull your desired image as you would do from any public registry:
 docker pull hub-public.libelektra.org/<image_name>:<tag_name>
 ```
 
+With podman:
```
Comment (markus2330) [inline doc/tutorials/run_all_tests_with_docker.md:84]:
```suggestion
Or with podman:
```

File: doc/tutorials/run_all_tests_with_docker.md [hunk 2]
```diff
@@ -65,12 +81,24 @@ Afterwards pull your desired image as you would do from any public registry:
...
+podman pull hub-public.libelektra.org/<image_name>:<tag_name>
+```
+
 Example:
 
 ```sh
 docker pull hub-public.libelektra.org/build-elektra-debian-stretch:201905-9dfe329fec01a6e40972ec4cc71874210f69933ab5f9e750a1c586fa011768ab
 ```
 
+With podman:
```
Comment (markus2330) [inline doc/tutorials/run_all_tests_with_docker.md:96]:
```suggestion
Or with podman:
```

**Review by markus2330 — APPROVED** [review]
Looks good to me!

File: doc/tutorials/run_all_tests_with_docker.md [hunk 3]
```diff
@@ -10,7 +10,7 @@ This is a step-by-step guide. Just follow the steps and you are good to go!
 
 ## Prerequisites
 
-- Docker for Linux containers has to be preinstalled. Please refer to https://docs.docker.com/install/ if you haven't installed it yet. Your host OS can be either Linux, macOS or Windows of course.
+- Docker for Linux containers has to be preinstalled. Please refer to https://docs.docker.com/install/ if you haven't installed it yet. Your host OS can be either Linux, macOS or Windows of course. Alternatively, you can use podman which is a different container engine, which is developed by RedHat. See https://podman.io/ for more details and an installation guide.
```
Comment (flo91) [inline doc/tutorials/run_all_tests_with_docker.md:13]:
```suggestion
- Docker for Linux containers has to be pre-installed. Please refer to https://docs.docker.com/install/ if you haven't installed it yet. Your host OS can be either Linux, macOS or Windows. Alternatively, you can use podman, a different container engine, which is developed by RedHat. See https://podman.io/ for more details and an installation guide.
```

**Review by flo91 — CHANGES_REQUESTED** [review]
Thank you for the PR!
That looks like a great improvement! :+1:
Currently two buildserver tests are still failing, but it looks like that's a general problem of the build system, as the same two checks also recently failed on many other PRs.


If the checklist for PRs is followed, two points of the category _Basics_ need attention:
- **Reference to issue**
	Most of the PRs for the FLOSS H2 task don't have references to issues.
	I'm not really sure if this point applies to our PRs.
	Should we create new issues for the changes we made?

- **The PR is rebased with current master.**
In https://www.libelektra.org/devgettingstarted/contributing#creating-a-pull-request is stated in point 6: _keep in mind to sync your fork - fetch, rebase & push_
When I open the repository for the PR (https://github.com/muskater/libelektra/tree/test_with_podman), there is the message _This branch is 5 commits ahead, 11 commits behind ElektraInitiative:master._ So it looks like this has still to be done.

File: doc/tutorials/run_all_tests_with_docker.md [hunk 4]
```diff
@@ -84,11 +112,60 @@ docker run -it --rm \
 buildelektra-sid
 ```
 
+#### 2.a Run the Docker Container with podman
+
+It is recommended that you run your container as a non-root user. As a result you have to set the correct permissions of the source directory.
```
Comment (Nirnberger) [inline doc/tutorials/run_all_tests_with_docker.md:117]:
```suggestion
It is recommended that you run your container as a non-root user. As a result, you have to set the correct permissions of the source directory.
```

File: doc/tutorials/run_all_tests_with_docker.md [hunk 5]
```diff
@@ -84,11 +112,60 @@ docker run -it --rm \
...
+-rw-r--r--. 1 root root   142  3. Nov 13:08 .oclint
+-rw-r--r--. 1 root root  9714  3. Nov 13:08 README.md
+-rw-r--r--. 1 root root   643  3. Nov 13:08 .restyled.yaml
+drwxr-xr-x. 1 root root     8  3. Nov 13:08 .reuse
+drwxr-xr-x. 1 root root   956  3. Nov 13:08 scripts
+drwxr-xr-x. 1 root root   110  3. Nov 13:08 src
+drwxr-xr-x. 1 root root   316  3. Nov 13:08 tests
+```
+
+Inside the source directory you can change the permissions to any user id with `podman unshare chown 1000:1000 -R .`. Keep in mind that this
```
Comment (Nirnberger) [inline doc/tutorials/run_all_tests_with_docker.md:151]:
```suggestion
Inside the source directory, you can change the permissions to any user id with `podman unshare chown 1000:1000 -R .`. Keep in mind that this
```

**Review by Nirnberger — APPROVED** [review]
A great addition to the documentation. Found two missing commas. Nice work :+1:

**Comment by markus2330** [comment]
> So it looks like this has still to be done.

Yes. And if you already did a rebase, the rebase doesn't seem to be perfect: there are some commits that are not yours in between.

**Comment by muskater** [comment]
Thank you a lot @Nirnberger and @flo91 for your suggestions. I have included them now in the PR.

**Comment by muskater** [comment]
I have removed the RedHat part from the tutorial, because I could not find a reliable source, that they are actually developing podman

**Comment by markus2330** [comment]
jenkins build libelektra please

**Comment by markus2330** [comment]
Thank you! :heart:

**Pull request merged**